### PR TITLE
A11y fixes

### PR DESCRIPTION
**Pull request by bstream**

### CODE REVIEW

File: src/components/labeled-value/labeled-value.jsx [hunk 1]
```diff
@@ -80,14 +90,14 @@ function LabeledValue({
   const labelClasses = cn(classes.label);
   const valueClasses = cn(classes.value, classes[`value-${size}`]);
 
+  const renderChild = setIdOnChild(id);
+
   return (
     <div {...rest} className={rootClasses}>
-      <span className={labelClasses} id={id}>
+      <label className={labelClasses} htmlFor={id}>
```
Comment (mjmonline) [inline src/components/labeled-value/labeled-value.jsx]: I'm not sure it's valid to use `<label>` if not together with a `<input />`

Comment (bstream) [reply src/components/labeled-value/labeled-value.jsx]: Good point! We probably need to re-think this some more... :)

Comment (JacobBlomgren) [reply src/components/labeled-value/labeled-value.jsx]: Yeah, the problem is that it most often used together with our `<Input />` that ends up rendering a `<input />`

**Comment by bstream** [comment]
This needs to be thought of some more, closing for now!

**Pull request closed without merging**